spec/associations_spec.cr

```diff
@@ -1,5 +1,11 @@
 require "./spec_helper"
 
+# class CredentialedUser < User
```

Comment:
Instead of inheriting from user, you can inherit from the model and only add a `has_one`. Not all fields need to be defined for every model, so that should work

Reply:
Sorry but I don't understand. Do you mean inherit from `LuckyRecord::Model`? Because that gives me the same issue as before: `SignInCredential::BaseQuery.new.credentialed_user_id(id).first`

Reply:
Ohh, I see what you mean now. I think the problem then, is that we need to add an option for setting the foreign_key: `has_one sign_in_credential : SignInCredential, foreign_key: :user_id`

But for now, I think the solution would be to create a different table (not users) and a different model. Maybe `Admin` and an `:admins` table. 

Does that make sense?

Reply:
Absolutely, I actually started adding a foreign_key option earlier but I thought I was complicating it. Good to know I was on the right track.

So now I'll add an Admin table where lets say credentials are required, and make the user credentials optional. Thus testing both?

Reply:
Yeah I think that makes the most sense 👍

Reply:
K I tried this too, but the method still looks for: `SignInCredential::BaseQuery.new.admin_id(id).first` so I either have to add both an `admin_id` and `user_id` which is a bad idea.

So I just added the `foreign_key` option to the `has_one` macro which wasn't difficult since I already added the option for `belongs_to`.